crypto/digest_extra/digest_test.cc

```diff
@@ -477,45 +478,93 @@ TEST(DigestTest, TransformBlocks) {
   EXPECT_TRUE(0 == OPENSSL_memcmp(ctx1.h, ctx2.h, sizeof(ctx1.h)));
 }
 
-// FIXME: Need to implement this for all hash functions used by HMAC
-TEST(DigestTest, InitAndGetStateSHA256) {
-  const size_t nb_blocks = 10;
-  const size_t block_size = SHA256_CBLOCK;
-  uint8_t data[block_size * nb_blocks];
-  for (size_t i = 0; i < sizeof(data); i++) {
-    data[i] = i*3;
+// DIGEST_TEST_InitAndGetState_Body expands to the body of the various
+// InitAndGetState* tests below.
+// Because EXPECT_* do not output the proper line when used in a macro,
```

Comment:
Maybe you can try `SCOPED_TRACE(hash1)` (or (ctx1)) https://github.com/google/googletest/blob/main/docs/advanced.md#adding-traces-to-assertions, not sure if it will help.

Reply:
The issue we have is that `EXPECT_*` would log the line where the macro is called/expanded, rather than the line in the definition of the macro. `SCOPED_TRACE` would unfortunately not helped in that case. I'm updating the comment in the code to make it clearer.